Sources/TensorFlow/Core/LazyTensorTFFunctionBuilder.swift

```diff
@@ -0,0 +1,324 @@
+                dims.withUnsafeMutableBufferPointer { dimsBuffer in
+                    ranks.withUnsafeBufferPointer { ranksBuffer in
+                        TF_SetAttrShapeList(
+                            desc,
+                            name,
+                            dimsBuffer.baseAddress,
+                            ranksBuffer.baseAddress,
+                            Int32(ranksBuffer.count))
+                    }
+                }
+            }
+        default: assert(false, "Unhandled attribute \(name):\(attrValue)")
+        }
+    }
+
+    private func makeTFGraphNode(
+        name: String,
+        attrs: [String: LazyTensorOperation.Attribute],
+        inputs: [Input],
+        device: String?
+    ) -> CTFOperation? {
+        // Create a new graph node now.
+        let desc: CTFOperationDescription! = TF_NewOperation(
+            cTFGraph,
+            name,
+            newNodeName(base: name))
+
+        // Set Attributes
+        for (name, value) in attrs {
+            updateAttribute(desc, name, value)
+        }
+
+        // Add Inputs
+        for input in inputs {
+            switch input {
+            case Input.single(let singleInput):
+                TF_AddInput(desc, singleInput)
+            case Input.list(let inputList):
+                inputList.withUnsafeBufferPointer { buffer in
+                    TF_AddInputList(desc, buffer.baseAddress, Int32(buffer.count))
+                }
+            }
+        }
+
+        if let device = device { TF_SetDevice(desc, device) }
+
+        // Finalize operation.
+        let graphNode = TF_FinishOperation(desc, status)
+        checkOk(status)
+        return graphNode!
+    }
+
+    private func makeTFConstNode(_ handle: TFETensorHandle) -> TF_Output {
+        let cTensorHandle = handle._cTensorHandle
+        let cTensor = TFE_TensorHandleResolve(cTensorHandle, status)
+        checkOk(status)
+        let desc = TF_NewOperation(cTFGraph, "Const", newNodeName(base: "Const"))
+        checkOk(status)
+        TF_SetAttrType(desc, "dtype", TFE_TensorHandleDataType(cTensorHandle))
+        TF_SetAttrTensor(desc, "value", cTensor, status)
+        checkOk(status)
+        let constNode = TF_FinishOperation(desc, status)
+        return TF_Output(oper: constNode, index: 0)
+    }
+
+    private func makeTFOutput(
+        _ lazyHandle: LazyTensor,
+        _ nodesCache: [ObjectIdentifier: CTFOperation?]
+    ) -> TF_Output {
+        if case let LazyTensor.Handle.symbolic(lazyOp, index, _) = lazyHandle.handle {
+            let id = ObjectIdentifier(lazyOp)
+            return TF_Output(oper: nodesCache[id]!, index: Int32(index))
+        }
+        assert(false, "Should only have symbolic inputs.")
+    }
+}
+
+/// Swift-side convenience wrapper for a `TF_Function`.
+class TFFunction {
+    let cTFFunction: CTFFunction
+    let outputCount: Int
+    let outputGroupCounts: [Int]
+
+    init(_ lazyTrace: LazyTensorTrace, name: String? = nil) {
```

Comment:
Initializers should have argument labels unless it is a trivial value-preserving conversion.
```suggestion
    init(trace: LazyTensorTrace, name: String? = nil) {
```